aioaseko/mobile.py

```diff
@@ -82,11 +96,41 @@ async def login(self) -> None:
             },
         )
         data = await resp.json()
-        self._access_token = data["accessToken"]
+        self.retrieve_tokens(data)
+
+    async def access_token(self) -> str | None:
+        """Return access token."""
+        now = time()
+        if (self.access_token_expiration is not None
+            and self.access_token_expiration <= now + TOKEN_EXPIRATION_OFFSET):
+            self._access_token = None
+            try:
+                if (self._refresh_token is not None):
+                    await self.refresh()
+            except InvalidAuthCredentials:
+                if (self._username is not None and self._password is not None):
+                    await self.login()
+        return self._access_token
```

Comment:
Changes:
- Try to login even if there is no refresh token
- Return string or raise `InvalidAuthCredentials`, never return None because this will fail in `_request()` where this is used.
- Rename to prevent confusion with `self._access_token`
- Added asserts for mypy typing

```suggestion
    async def _get_valid_access_token(self) -> str:
        """Return a valid access token."""
        if self._access_token is not None:
            assert self._access_token_expiration is not None
            if self._access_token_expiration >= time() + TOKEN_EXPIRATION_BUFFER:
                return self._access_token
        if self._refresh_token is not None:
            try:
                await self._refresh()
            except InvalidAuthCredentials:
                self._refresh_token = None
            else:
                assert self._access_token is not None
                return self._access_token
        if self._username is not None and self._password is not None:
            await self.login()
            assert self._access_token is not None
            return self._access_token
        raise InvalidAuthCredentials("No valid refresh token or username and password available.")
```